test/gtest/include/common/view-tests.hpp

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+TYPED_TEST_P(CommonTests, Subrange) {
+  using DV = typename TypeParam::DV;
+  using V = typename TypeParam::V;
+
+  std::size_t n = 10;
+
+  auto neg = [](auto &v) { v = -v; };
+  DV dv_a(n);
+  TypeParam::iota(dv_a, 100);
+  TypeParam::for_each(TypeParam::policy(),
+                      rng::subrange(dv_a.begin() + 1, dv_a.end() - 1), neg);
+
+  if (comm_rank == 0) {
+    V a(n), a_in(n);
+    rng::iota(a, 100);
+    rng::iota(a_in, 100);
+    rng::for_each(rng::subrange(a.begin() + 1, a.end() - 1), neg);
+    EXPECT_TRUE(unary_check(a_in, a, dv_a));
```

Comment:
Check look perfectly readable when having exact values which we expect, like
```
EXPECT_TRUE(equal(dist_vec, V{2, 3, 4}));
```
... so one has no to trace what happened to to a_in, a, dv_a during the test and what does `unary_check` do.
See [slice unittest](https://github.com/ericniebler/range-v3/blob/master/test/view/slice.cpp) for a good example on how checks on expected results can look like.

Reply:
check_equal and unary_check are similar, I don't see a difference.

Reply:
`equal` is standard and has obvious semantic, reader has to see `unary_check` for implementation to see that 1st param is just for printing it out and 2nd, 3rd are for equality check

Reply:
I don't object if you want to port or adapt the range v3 tests or write tests in that style. I started writing the DR tests that way and found they weren't what I needed to find and fix the bugs. There are complex interactions between views and algorithms, off by 1 and other boundary conditions and I need to write tests that are flexible and combine views and algorithms to find the issues.